include/openPMD/RecordComponent.hpp

```diff
@@ -170,7 +134,10 @@ RecordComponent::loadChunk(Offset const& o, Extent const& e, std::unique_ptr< T[
         numPoints *= dimensionSize;
 
     if( (Allocation::AUTO == alloc && !data) || Allocation::API == alloc )
-        data = std::unique_ptr< T[], D >(new T[numPoints]);
+    {
+        auto newData = std::shared_ptr<T>( new T[numPoints], []( T *p ){ delete [] p; } );
+        data.swap(newData);
```

Comment:
Are you doing this for safety guarantee when the allocation would fail?

Reply:
I was trying to exchange the incoming shared_ptr but things did not work as expected and I started to go for the two-signature option for allocations...